dref/serializers.py

```diff
@@ -0,0 +1,220 @@
+            for district in districts:
+                if district.country != data['country']:
+                    raise serializers.ValidationError({
+                        'district': ugettext('Different districts found for given country')
+                    })
+        return data
+
+
+class DrefSerializer(
+    EnumSupportSerializerMixin,
+    NestedUpdateMixin,
+    NestedCreateMixin,
+    ModelSerializer
+):
+    MAX_NUMBER_OF_IMAGES = 6
+    ALLOWED_BUDGET_FILE_EXTENSIONS = ["pdf"]
+
+    country_district = DrefCountryDistrictSerializer(source='drefcountrydistrict_set', many=True, required=False)
+    national_society_actions = NationalSocietyActionSerializer(many=True, required=False)
+    needs_identified = IdentifiedNeedSerializer(many=True, required=False)
+    planned_interventions = PlannedInterventionSerializer(many=True, required=False)
+    type_of_onset_display = serializers.CharField(source='get_type_of_onset_display', read_only=True)
+    disaster_category_display = serializers.CharField(source='get_disaster_category_display', read_only=True)
+    status_display = serializers.CharField(source='get_status_display', read_only=True)
+    modified_by_details = UserNameSerializer(source='modified_by', read_only=True)
+    event_map_details = DrefFileSerializer(source='event_map', read_only=True)
+    images_details = DrefFileSerializer(source='images', many=True, read_only=True)
+    field_report_details = MiniFieldReportSerializer(source='field_report', read_only=True)
+    created_by_details = UserNameSerializer(source='created_by', read_only=True)
+    users_details = UserNameSerializer(source='users', many=True, read_only=True)
+    budget_file_details = DrefFileSerializer(source='budget_file', read_only=True)
+    cover_image_details = DrefFileSerializer(source='cover_image', read_only=True)
+    disaster_type_details = DisasterTypeSerializer(source='disaster_type', read_only=True)
+
+    class Meta:
+        model = Dref
+        fields = '__all__'
+        read_only_fields = ('modified_by', 'created_by', 'budget_file_preview')
+
+    def remove_digits_after_decimal(self, field):
+        # NOTE: We are doing this to remove decimal after 3 digits whole numbers
+        # eg: 100.00% be replaced with 100%
+        if field and len(field.split('.')[0]) == 3:
+            return field.split('.')[0]
+        return None
+
+    def to_representation(self, instance):
+        data = super().to_representation(instance)
+        disability_people_per = data.get('disability_people_per', '')
+        people_per_urban = data.get('people_per_urban', '')
+        people_per_local = data.get('people_per_local', '')
+        data['disability_people_per'] = self.remove_digits_after_decimal(disability_people_per)
+        data['people_per_urban'] = self.remove_digits_after_decimal(people_per_urban)
+        data['people_per_local'] = self.remove_digits_after_decimal(people_per_local)
+        return data
```

Comment:
Continuation of https://github.com/IFRCGo/go-api/pull/1141#discussion_r798225677

```suggestion
    def to_representation(self, instance):
        def _remove_digits_after_decimal(data, key):
            # NOTE: We are doing this to remove decimal after 3 digits whole numbers
            # eg: 100.00% be replaced with 100%
            if value and len(value.split('.')[0]) == 3:
                return value.split('.')[0]
            return None

        data = super().to_representation(instance)
        for key in [
            'disability_people_per',
            'people_per_urban',
            'people_per_local',
        ]:
            value = data.get(key) or ''
            data[key] = _remove_digits_after_decimal(value)
        return data
```